src/metrics/representation.py

```diff
@@ -0,0 +1,381 @@
+        
+        Firstly it gather all storage.
+        Prepare data, separate queries vectors and data base vectors.
+        Then build faiss index.
+        Create generator of relevant, closest arrays.
+
+        Returns:
+            Generator wich return relevant and closest examples.
+        """
+        vectors = torch.cat(self.vectors).numpy()
+        
+        if self.normalize_input:
+            vectors = normalize(vectors)
+
+        if self.dataset == 0:
+            # if classification dataset
+            targets = torch.cat(self.targets).numpy()
+            # prepare data
+            q_vecs, db_vecs, relevants, scores, \
+                db_idxs, q_order_idxs = self.prepare_classification_data(vectors, targets)
+        else:
+            # if representation dataset
+            scores = torch.cat(self.scores).numpy()
+            # convert numpy array with type long to bool
+            queries_idxs = torch.cat(self.queries_idxs).numpy()
+            # prepare data
+            q_vecs, db_vecs, relevants, scores, \
+                db_idxs, q_order_idxs = self.prepare_representation_data(vectors, queries_idxs, scores)
+
+        q_vecs = q_vecs.astype(np.float32)
+        db_vecs = db_vecs.astype(np.float32)
+
+        index = self.build_index(db_vecs)
+
+        # if search batch size is None, search queries vectors by one request
+        search_batch_size = len(q_vecs) if self.search_batch_size is None else self.search_batch_size
+
+        # if k is None set it as database length
+        k = len(db_vecs) if self.k is None else self.k
+
+        generator = self.query_generator(index, relevants, q_vecs, scores, db_idxs, q_order_idxs, search_batch_size, k)
+        return generator
+
+    def prepare_representation_data(self, vectors: np.ndarray, queries_idxs: np.ndarray, scores: np.ndarray
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        """Prepare data for faiss build index, and following search, in representation dataset case.
+        
+        Separate query and database vectors from storage vectors.
+        Prepare scores.
+        Generate relevant indexes for every query request.
+
+        Args:
+            vectors: Vectors of all storage, include queries and database vectors, size (database_size, embedding_size).
+            queries_idxs: Bool array wich indicate if vector is query, size (database_size).
+            scores: Array of scores for every relevant index per each query request, size (database_size, query_size).
+                See representation dataset for more information.
+
+        Retruns:
+            q_vecs: Queries vectors, size (queries_size, embedding_size).
+            db_vecs: Database vectors, for faiss build index, size (database_size, embedding_size).
+            relevant: Array of relevant indexes in database for every query request, size (queries_size, ).
+            scores: Array of scores without queries empty scores.
```

Comment:
"scores: Array of scores related to queries that have at least one relevant item."